internal/config/read_test.go

```diff
@@ -95,6 +99,49 @@ func TestConfigReadFromEnv_NotInteger(t *testing.T) {
 	}
 
 	if len(invConfigError) != 2 {
+		t.Errorf("%v", invConfigError)
 		t.Errorf("config.readFromEnv() error length = %d, want 2", len(invConfigError))
 	}
 }
+
+func TestConfigReadFromEnv_MissingBuildId(t *testing.T) {
+	os.Setenv("BUILDKITE_SPLITTER_BASE_URL", "")
+	os.Setenv("BUILDKITE_SPLITTER_MODE", "")
+	os.Setenv("BUILDKITE_SPLITTER_TEST_CMD", "")
+	os.Setenv("BUILDKITE_SPLITTER_RETRY_COUNT", "")
+	os.Setenv("BUILDKITE_STEP_ID", "123")
+	defer os.Clearenv()
+
+	c := Config{}
+	err := c.readFromEnv()
+
+	var invConfigError InvalidConfigError
+	want := "BUILDKITE_BUILD_ID must not be blank"
+
+	if errors.As(err, &invConfigError) {
```

Comment:
If the error is not an `InvalidConfigError`, this test will pass

Reply:
How's that?

Reply:
Because if the `err` is not an `InvalidConfigError`, the `t.Errorf` within will not be called, and the test function will finish without throwing any failure.

Now that you've added the check above, you can remove the `if` condition here.

```
var invConfigError InvalidConfigError
if !errors.As(err, &invConfigError) {
	t.Fatalf("config.readFromEnv() error = %v, want InvalidConfigError", err)
}

if got := invConfigError[0].Error(); got != want {
	t.Errorf("config.readFromEnv() got = %v, want = %v", got, want)
}
```

Notice that I use `t.Fatalf` on the first check to prevent test function to execute the next check.
You can refer to this [test](https://github.com/buildkite/test-splitter/blob/6b67efc34acfbca72bbfe69c904aabae4152f6e6/internal/runner/command_test.go#L26) implementation.

Reply:
Oh sorry, by how's that, I meant how does this commit look: https://github.com/buildkite/test-splitter/pull/175/commits/b468111e35f766011f598b28c87207519e4359d5

Reply:
ah! 😄